File: p4/deepspace/src/deepspace/PowerEfficientSpaceStation.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

import java.util.ArrayList;

/**
 *
 * @author Sergio
 */
class PowerEfficientSpaceStation extends SpaceStation{
    
    private static final float EFFICIENCYFACTOR = 1.10f;
    
    PowerEfficientSpaceStation(SpaceStation station){             
        super(station);       
    }
    
    @Override
    public float fire(){
        return super.fire()*EFFICIENCYFACTOR;
    }
    
    @Override
    public float protection(){
        return super.protection()*EFFICIENCYFACTOR;
    }
    @Override
    public Transformation setLoot(Loot loot){
       Transformation transformation = super.setLoot(loot);
       if(transformation == Transformation.SPACECITY)
           transformation = Transformation.NOTRANSFORM;
       return transformation;
   
    }
    
    public PowerEfficientSpaceStationToUI getUIversion(){
        return new PowerEfficientSpaceStationToUI(this);
    }
    
}

File: p4/Deepspace/PowerEfficientSpaceStation.rb
#encoding:utf-8
require_relative "PowerEfficientSpaceStationToUI.rb"
require_relative "SpaceStation.rb"
require_relative "SuppliesPackage.rb" #ABORRAR
module Deepspace
    class PowerEfficientSpaceStation < SpaceStation
        @@EFFICIENCYFACTOR = 1.1
        def initialize(station)
            super(station.name(),SuppliesPackage.new(station.ammoPower(),station.fuelUnits(),station.shieldPower()))
            @nMedals = station.nMedals
				
	    @hangar = station.hangar
            @pendingDamage = station.pendingDamage
	    @shieldBoosters = station.shieldBoosters 
	    @weapons = station.weapons 
        end
        def fire()
            super*@@EFFICIENCYFACTOR
        end
        def protection()
            super*@@EFFICIENCYFACTOR
        end
        def setLoot(loot)
            trans = super(loot);
            if (trans == Transformation::SPACECITY)
                return Transformation::NOTRANSFORM;
            else
                return trans;
            end 
        end
        def getUIversion()
			Deepspace::PowerEfficientSpaceStationToUI.new(self)
	end
    end
    #station = SpaceStation.new("p",SuppliesPackage.new(1,2,3))
    #puts station.fire()
    #puts station.protection()
    #pwsp = PowerEfficientSpaceStation.new(station)
    #puts pwsp.fire()
    #puts pwsp.protection()
end
